apps/dot_ext/tests/test_form_application.py

```diff
@@ -258,33 +258,6 @@ def test_update_form_edit(self):
         form = CustomRegisterApplicationForm(user, passing_app_fields)
         self.assertTrue(form.is_valid())
 
-        # Test client_type = 'confidential' and authorization_grant_type = 'implicit' not allowed.
```

Comment:
Wondering if we should be removing these tests or just reworking what goes on here. It could be a good test to still verify that regardless of what the form data says, we will save `confidential` and `authorization-code`. Would it make sense for the `clean` function to save those values to these fields in the `cleaned_data` and then to adjust these tests to instead of expecting an error, to expect that the `cleaned_data` shows what we want it to regardless of what was passed in? Part of that question is: should the values be hardcoded in `clean`, `save`, or both? @loganbertram hoping you can weigh in here too.

Reply:
Yeah, I'd leave lines 89-90 in clean() and drop setting the values in save(). I feel like the "cleaning" being done is setting legacy values for fields that don't actually accept other inputs.

Reply:
I think pulling the tests out makes sense. No way to change these values now and, in the future, if we accept other values, we would need to change this test anyway. I'd be ok with minimal changes too, but this whole testing mechanism doesn't need to exist.

Reply:
@loganbertram I tried to move this logic to the clean method but it did not save the fields back to the database.
@jimmyfagan @loganbertram I think we can leave as is for now and do share your thoughts on it?